coreplugins/cesiumion/public/TaskView.jsx

```diff
@@ -0,0 +1,287 @@
+				this.cancelableFetch.cancel();
+				this.cancelableFetch = null;
+			} catch (exception) {
+				console.warning("Failed to clear dead requests");
+				console.warning(exception);
+			}
+		}
+
+		this.refreshAssets = null;
+	}
+
+	render() {
+		const { task, token } = this.props;
+		const {
+			isTasksDialog,
+			isUploadDialogLoading,
+			isRefreshTask,
+			currentAsset
+		} = this.state;
+		const isUploadDialog = currentAsset !== null;
+		const assetName = isUploadDialog ? AssetStyles[currentAsset].name : "";
+
+		return (
+			<AppContext.Provider value={this.props}>
+				<ErrorMessage bind={[this, "error"]} />
+				<div className={"ion-dropdowns"}>
+					<TaskFetcher
+						path={"share"}
+						onLoad={this.onAssetsRefreshed}
+						onBindRefresh={method => (this.refreshAssets = method)}
+					>
+						{({ isError, data = {} }) => {
+							// Asset Export and View Selector
+							const { items = [] } = data;
+							const available = items
+								.filter(
+									item =>
+										(!item.isExported && !item.isTask) ||
+										item.isError
+								)
+								.map(item => item.type);
+							const exported = items
+								.filter(item => item.isExported)
+								.map(item => item.type);
+
+							// Tasks Selector
+							const processing = items.filter(
+								item => item.isTask
+							);
+							const isTasks = processing.length > 0;
+							const isErrors = processing.some(
+								item => item.isError
+							);
+
+							return (
+								<Fragment>
+									{available.length > 0 && (
+										<IonAssetButton
+											assets={available}
+											onSelect={this.onOpenUploadDialog}
+										>
+											Tile in Cesium Ion
+										</IonAssetButton>
+									)}
+
+									{exported.length > 0 && (
+										<IonAssetButton
+											assets={exported}
+											onSelect={this.handleAssetSelect(
+												data
+											)}
+										>
+											View in Cesium Ion
+										</IonAssetButton>
+									)}
+									{items.length <= 0 && (
+										<button
+											className={"ion-btn btn btn-primary btn-sm"}
+											onClick={this.refreshAssets}
+										>
+											<i className={"fa fa-cesium"} />
+											Refresh Available ion Assets
+										</button>
+									)}
+									{isTasks && (
+										<button
+											className={`ion-btn btn btn-sm ${isErrors ? "btn-danger" : "btn-primary"}`}
+											onClick={this.showTaskDialog}
+										>
+											<i className={"fa fa-cesium"} />
+											View ion Tasks
+										</button>
+									)}
+									<TasksDialog
+										title={"Cesium ion Tasks"}
+										show={isTasksDialog}
+										tasks={processing}
+										onHide={this.hideTaskDialog}
+										onClearFailed={this.onClearFailedAssets}
+										ref={(domNode) => { this.tasksDialog = domNode; }}
+									/>
+								</Fragment>
+							);
+						}}
+					</TaskFetcher>
+				</div>
+
+				<APIFetcher path={"projects/"} params={{ id: task.project }}>
```

Comment:
I don't love this APIFetcher pattern. I don't think the network requests should be coupled to a component. Seems like you put a decent amount of work into though so ignore me if too much work